runtime/vm/jnicsup.cpp

```diff
@@ -468,14 +464,27 @@ UDATA JNICALL   pushArguments(J9VMThread *vmThread, J9Method* method, void *args
 		switch (*sigChar++) {
 			case '[':
 				/* skip the rest of the signature */
-				while (*sigChar == '[') sigChar++;
-				if (*sigChar++ != 'L') goto dontSkipClassName;
+				while (*sigChar == '[') {
+					sigChar++;
+				}
+				if ('L' == *sigChar++) {
+					/* skip the rest of the signature */
+					while (';' != *sigChar) {
+						sigChar += 1;					
+					}
+				}
+				sp -= 1;
+				objArg = ARG(jobject, l);
+				*sp = (0 == objArg) ? 0: (UDATA) *((j9object_t*) objArg);
+				break;
 			case 'L':
 				/* skip the rest of the signature */
-				while (*sigChar++ != ';');
-dontSkipClassName:
+				while (';' != *sigChar) {
+					sigChar += 1;
+				}
+				sp -= 1;
 				objArg = ARG(jobject, l);
-				*--sp = objArg ? (UDATA)*(j9object_t*)objArg : 0;
+				*sp = (0 == objArg) ? 0: (UDATA) *((j9object_t*) objArg);
```

Comment:
NULL rather than 0 for the comparison.

Reply:
Hi, I'm an user of J9 and encountered one issue on jdk80 64bit sr6fp35 occurred in routine pushArguments()  in jnicsup.cpp and seems to be related to the updated line of code here. would you kindly help to review?
The issue was:
When I set  "_export MALLOCDEBUG=catch_overflow,align:0_" to enable memory catch-overflow and a simple Hello java program would crash on the line:  _*sp = (NULL == objArg) ? 0: (UDATA) *((j9object_t*) objArg);_

Several findings by lots of tests：
1, besides my using sr6fp35, many other older and newer versions on jdk8064bit had the issue as well.
2, 32bit java did not have the issue.
3, when option _align_ in MALLOCDEBUG was changed to be 8 or 16 or bigger, the issue did not occur.

I tried to understand the line of code but did not find out obvious overflow. So request your help. Thanks.

Reply:
@yongqinglu I no longer maintain this code.
If you think there is an problem, please open a github issue.
Thank you
             -p

Reply:
Using `align:0`, which really means `align:1`, is invalid / incompatible with OpenJ9 and expected to cause problems. OpenJ9 64-bit expects the allocated memory to be aligned as per the default 2 word boundary. Changing the default to a smaller alignment will cause undefined results.

Reply:
Thank you all. Per my understanding of your input, it worked as designed and was not an issue. Thanks. @pshipton